icloudpd: init at 1.17.3

Enzime opened the pull request:
## Description of changes

Replaces #171998 and #246307 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Enzime committed:
icloudpd: init at 1.17.3

kirillrdy commented on pkgs/by-name/ic/icloudpd/package.nix:
```diff
@@ -0,0 +1,57 @@
+python3Packages.buildPythonPackage rec {
+  pname = "icloudpd";
+  version = "1.17.3";
+  format = "pyproject";
```
```suggestion
  pyproject = true;
```

kirillrdy commented on pkgs/by-name/ic/icloudpd/package.nix:
```diff
@@ -0,0 +1,57 @@
+    owner = "icloud-photos-downloader";
+    repo = "icloud_photos_downloader";
+    rev = "v${version}";
+    sha256 = "sha256-GS6GqlZfj5kfjKLImkOTDAgQDGJQHl74uTqbZHVpbac=";
```
```suggestion
    hash = "sha256-GS6GqlZfj5kfjKLImkOTDAgQDGJQHl74uTqbZHVpbac=";
```

kirillrdy commented on pkgs/by-name/ic/icloudpd/package.nix:
```diff
@@ -0,0 +1,57 @@
+, testers
+}:
+
+python3Packages.buildPythonPackage rec {
```
```suggestion
python3Packages.buildPythonApplication rec {
```

kirillrdy commented on pkgs/by-name/ic/icloudpd/package.nix:
```diff
@@ -0,0 +1,57 @@
+
+  preBuild = ''
+    substituteInPlace pyproject.toml \
+      --replace "setuptools==69.0.2" "setuptools" \
```
consider using pythonRelaxDepsHook

Enzime replied:
I tried using it, but I don't think it works as the specific versions of `setuptools` and `wheel` are specified as in build system section

https://github.com/icloud-photos-downloader/icloud_photos_downloader/blob/master/pyproject.toml#L1-L5

kirillrdy commented on pkgs/by-name/ic/icloudpd/package.nix:
```diff
@@ -0,0 +1,57 @@
+    homepage = "https://github.com/icloud-photos-downloader/icloud_photos_downloader";
+    description = "iCloud Photos Downloader";
+    license = licenses.mit;
+    maintainers = with maintainers; [ anpin Enzime ];
```
please add `mainProgram`

kirillrdy commented on pkgs/by-name/ic/icloudpd/package.nix:
```diff
@@ -0,0 +1,57 @@
+      --replace "wheel==0.42.0" "wheel"
+  '';
+
+  doCheck = false;
```
no tests ?

kirillrdy commented:
also fails to build 
```
mport-from-derivation --option build-use-sandbox relaxed -f /home/kirillvr/.cache/nixpkgs-review/pr-280643/build.nix
error: builder for '/nix/store/va9xv1m8ljhsh6m2wn9l5h81nn0l2n5b-python3.11-icloudpd-1.17.3.drv' failed with exit code 1;
       last 10 log lines:
       > Executing pythonRuntimeDepsCheck
       > Checking runtime dependencies for icloudpd-1.17.3-py3-none-any.whl
       >   - click==8.1.6 not satisfied by version 8.1.7
       >   - tqdm==4.66.0 not satisfied by version 4.66.1
       >   - urllib3==1.26.16 not satisfied by version 2.1.0
       >   - tzlocal==4.3.1 not satisfied by version 5.2
       >   - pytz==2022.7.1 not satisfied by version 2023.3.post1
       >   - certifi==2022.12.7 not satisfied by version 2023.11.17
       >   - keyring==23.13.1 not satisfied by version 24.3.0
       >   - keyrings-alt==4.2.0 not satisfied by version 5.0.0
       For full logs, run 'nix log /nix/store/va9xv1m8ljhsh6m2wn9l5h81nn0l2n5b-python3.11-icloudpd-1.17.3.drv'.
error: 1 dependencies of derivation '/nix/store/08blawpps880zc4iszy4ipyf81hlp909-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/280643

2 packages failed to build:
icloudpd icloudpd.dist
```

Enzime commented:
@kirillrdy I've updated my PR and added tests

This commit builds when it's on top of `nixos-unstable`, but it doesn't on top of `master`, which I've fixed by adding `pythonRelaxDepsHook`.

kirillrdy reviewed (approved):
Result of `nixpkgs-review pr 280643` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>icloudpd</li>
    <li>icloudpd.dist</li>
  </ul>
</details>

anpin reviewed (approved):
LGTM
